src/shmem_env_defs.h

```diff
@@ -91,7 +91,11 @@ SHMEM_INTERNAL_ENV_DEF(OFI_TX_POLL_LIMIT, long, DEFAULT_POLL_LIMIT, SHMEM_INTERN
 SHMEM_INTERNAL_ENV_DEF(OFI_RX_POLL_LIMIT, long, DEFAULT_POLL_LIMIT, SHMEM_INTERNAL_ENV_CAT_TRANSPORT,
                        "Get completion poll limit")
 SHMEM_INTERNAL_ENV_DEF(OFI_STX_MAX, long, 1, SHMEM_INTERNAL_ENV_CAT_TRANSPORT,
-                       "Maximum number of STX contexts")
+                       "Maximum number of STX contexts per PE")
+SHMEM_INTERNAL_ENV_DEF(OFI_STX_NODE_MAX, long, -1, SHMEM_INTERNAL_ENV_CAT_TRANSPORT,
+                       "Maximum number of STX contexts per compute node")
+SHMEM_INTERNAL_ENV_DEF(OFI_STX_AUTO, bool, false, SHMEM_INTERNAL_ENV_CAT_TRANSPORT,
+                       "Automatically partition the STX pool evenly across PEs on each compute node")
```

Comment:
I would say yes, unless we change all the other `bool` values in this file to behave that way as well.  Should we create an issue for this?  I'd say it's ok, but the readme file should be more clear about this.

Reply:
Sorry, I definitely misread your original question here (I was thinking of whether `OFI_STX_AUTO` should or shouldn't be enabled with a negative value).

To answer your actual question, I think it would be good to have `OFI_STX_MAX = -1` mean auto, but what about all other negative values?  At the moment, if `SHMEM_THREAD_MULTIPLE` is on and/or `STX_MAX` is negative (and `STX_AUTO` is off), then things break.  I'm wondering whether any negative value for `STX_MAX` should enable `STX_AUTO`, or if only -1  should (then throw in an assert to catch other negative numbers before bad things happen).  It seems a little simpler to do the former, but I'm open to the latter too.

Reply:
Actually things break in a similar way if `STX_MAX` is 0 as well...

Reply:
I ended up going with any non-positive value for `OFI_STX_MAX` enabling `OFI_STX_AUTO`.